docs/reference/schemas/config/functions/tryGet.md

```diff
@@ -0,0 +1,312 @@
+environments without causing errors. This example demonstrates retrieving
+feature flags that may or may not be defined. This example uses
+[`createObject()`][03] to build the configuration object and [`coalesce()`][02]
+for fallback values.
+
+```yaml
+# tryGet.example.1.dsc.config.yaml
+$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
+resources:
+- name: Feature Flags
+  type: Microsoft.DSC.Debug/Echo
+  properties:
+    output:
+      config: "[createObject('enableBeta', true, 'enableDebug', false)]"
+      betaEnabled: "[coalesce(tryGet(createObject('enableBeta', true, 'enableDebug', false), 'enableBeta'), false)]"
+      alphaEnabled: "[coalesce(tryGet(createObject('enableBeta', true, 'enableDebug', false), 'enableAlpha'), false)]"
+      debugEnabled: "[tryGet(createObject('enableBeta', true, 'enableDebug', false), 'enableDebug')]"
+```
+
+```bash
+dsc config get --file tryGet.example.1.dsc.config.yaml
+```
+
+```yaml
+results:
+- name: Feature Flags
+  type: Microsoft.DSC.Debug/Echo
+  result:
+    actualState:
+      output:
+        config:
+          enableBeta: true
+          enableDebug: false
+        betaEnabled: true
+        alphaEnabled: false
+        debugEnabled: false
+messages: []
+hadErrors: false
+```
+
+The function safely returns `true` for the existing `enableBeta` flag, `null`
+for the non-existent `enableAlpha` flag (which `coalesce()` converts to
+`false`), and `false` for the `enableDebug` flag.
+
+### Example 2 - Access deployment environment settings
+
+Retrieve environment-specific configuration values with safe defaults. This is
+useful when different environments might have different configuration keys. This
+example uses [`createObject()`][03] to build the environment settings and
+demonstrates accessing nested object properties.
+
+```yaml
+# tryGet.example.2.dsc.config.yaml
+$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
+resources:
+- name: Environment Config
+  type: Microsoft.DSC.Debug/Echo
+  properties:
+    output:
+      environments: "[createObject('production', createObject('replicas', 5, 'region', 'us-east-1'), 'staging', createObject('replicas', 2))]"
+      productionEnv: "[tryGet(createObject('production', createObject('replicas', 5, 'region', 'us-east-1'), 'staging', createObject('replicas', 2)), 'production')]"
+      stagingEnv: "[tryGet(createObject('production', createObject('replicas', 5, 'region', 'us-east-1'), 'staging', createObject('replicas', 2)), 'staging')]"
+      developmentEnv: "[tryGet(createObject('production', createObject('replicas', 5, 'region', 'us-east-1'), 'staging', createObject('replicas', 2)), 'development')]"
+      prodReplicas: "[tryGet(createObject('replicas', 5, 'region', 'us-east-1'), 'replicas')]"
+      prodRegion: "[tryGet(createObject('replicas', 5, 'region', 'us-east-1'), 'region')]"
+      stagingRegion: "[tryGet(createObject('replicas', 2), 'region')]"
```

Comment:
Although technically this works, I think as an example, it's easier for the user to understand objects defined as parameters as they don't need to mentally construct it.